subsys/net/lib/dhcpv4/dhcpv4_server.c

```diff
@@ -472,6 +489,13 @@ static uint8_t *dhcpv4_encode_requested_params(
 			}
 			break;
 
+		case DHCPV4_OPTIONS_ROUTER:
+			buf = dhcpv4_encode_router_option(
+				buf, buflen, &ctx->iface->config.ip.ipv4->gw);
+			if (buf == NULL) {
+				goto out;
+			}
+			break;	
```

Comment:
If you are using vscode, have a look at [editorconfig](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig) & [checkpatch](https://github.com/ycsin/zephyrproject-extension-pack/tree/pr/update_readme_0?tab=readme-ov-file#checkpatch) to minimize this kind of stuff